Lib/fontTools/ttLib/tables/O_S_2f_2.py

```diff
@@ -508,11 +506,20 @@ def intersectUnicodeRanges(unicodes, inverse=False):
 	True
 	"""
 	unicodes = set(unicodes)
-	uniranges = _getUnicodeRangeSets()
-	bits = set([
-		bit for bit, unirange in enumerate(uniranges)
-		if not unirange.isdisjoint(unicodes) ^ inverse])
-	return bits
+	unicodestarts, unicodevalues = _getUnicodeRanges()
+	bits = set()
+	for code in unicodes:
+		stop, bit = unicodevalues[bisect.bisect(unicodestarts, code)]
```

Comment:
This will raise `TypeError` if `code` is negative (`unicodevalues[0]` being `None`). Normally this is of course not possible – I'm just wondering how defensive this code needs to be.

Reply:
The real question here is: what do we want to do when this happens. If we want to silently ignore negative values, the fastest solution I can think of is to replace `None` by `(None, None)` in `unicodevalues`, and to remove `None` from `bits`. Or we can just ignore them in the loop if we prefer readability (it’s about 3% slower with the 1000 values of the testing script).

Reply:
I think it's acceptable to raise TypeError on negative codepoint, which is invalid and unlikely in normal circumstances.

Reply:
> I think it's acceptable to raise TypeError on negative codepoint, which is invalid and unlikely in normal circumstances.

Then OK, let’s keep this.